linode_api4/linode_client.py

```diff
@@ -1433,6 +1433,513 @@ def cancel(self):
+        :rtype: PaginatedList of ObjectStorageBucket
+        """
+        return self.client._get_and_filter(ObjectStorageBucket, *filters)
+
+    def bucket_create(
+        self,
+        cluster,
+        label,
+        acl: ObjectStorageACL = ObjectStorageACL.PRIVATE,
+        cors_enabled=False,
+    ):
+        """
+        Creates an Object Storage Bucket in the specified cluster. Accounts with
+        negative balances cannot access this command. If the bucket already exists
+        and is owned by you, this endpoint returns a 200 response with that bucket
+        as if it had just been created.
+
+        This endpoint is available for convenience.
+        It is recommended that instead you use the more fully-featured S3 API directly.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-bucket-create
+
+        :param acl: The Access Control Level of the bucket using a canned ACL string.
+                    For more fine-grained control of ACLs, use the S3 API directly.
+        :type acl: str
+                   Enum: private,public-read,authenticated-read,public-read-write
+
+        :param cluster: The ID of the Object Storage Cluster where this bucket
+                        should be created.
+        :type cluster: str
+
+        :param cors_enabled: If true, the bucket will be created with CORS enabled for
+                             all origins. For more fine-grained controls of CORS, use
+                             the S3 API directly.
+        :type cors_enabled: boolean
+
+        :param label: The name for this bucket. Must be unique in the cluster you are
+                      creating the bucket in, or an error will be returned. Labels will
+                      be reserved only for the cluster that active buckets are created
+                      and stored in. If you want to reserve this bucket’s label in
+                      another cluster, you must create a new bucket with the same label
+                      in the new cluster.
+        :type label: str
+
+        :returns: A Object Storage Buckets that created by user.
+        :rtype: ObjectStorageBucket
+        """
+        cluster_id = (
+            cluster.id if isinstance(cluster, ObjectStorageCluster) else cluster
+        )
+
+        params = {
+            "cluster": cluster_id,
+            "label": label,
+            "acl": acl,
+            "cors_enabled": cors_enabled,
+        }
+
+        result = self.client.post("/object-storage/buckets", data=params)
+
+        if not "label" in result or not "cluster" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when creating Object Storage Bucket!",
+                json=result,
+            )
+
+        return ObjectStorageBucket(
+            self.client, result["label"], result["cluster"], result
+        )
+
+    def buckets_in_cluster(self, cluster_id, *filters):
```

Comment:
Maybe this could be implemented as a method under the `ObjectStorageCluster` class

Reply:
Cool, I can move it. Just curious, what's the philosophy of where a method should be? Because sometimes (maybe most of the time lol), I'm not sure if a method should under a specific class or the group.